Regenerates QuantumEngineService client libraries to include GetQuant…

hoisinberg opened the pull request:
…umProcessorConfig

hoisinberg commented:
Additionally tested locally with a colab which sent jobs to the production QuantumEngineService and retrieved results.

mhucka commented:
@hoisinberg Some best-practices related requests:

* Please avoid splitting titles mid-word. It's best if PR and issue titles are self-contained, so that someone reading long lists of them (dozens or hundreds) can figure out more or less what they will do.
* Please add a more comprehensive description to this PR. Descriptions should explain the context, the purpose of the pull request, and the effects of the PR.

mhucka commented:
> Additionally tested locally with a colab which sent jobs to the production QuantumEngineService and retrieved results.

Is this a comment about what the PR does, or an instruction to maintainers, or something else?

hoisinberg commented:
> > Additionally tested locally with a colab which sent jobs to the production QuantumEngineService and retrieved results.
> 
> Is this a comment about what the PR does, or an instruction to maintainers, or something else?

It was a description of some additional testing in order to verify that the updated RPC clients can actually communicate with the production QuantumEngionService. I've fixed up the wording and added it to the PR description as well.

pavoljuhas commented on cirq-google/cirq_google/cloud/quantum/__init__.py:
```diff
@@ -13,6 +13,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from cirq_google.cloud.quantum import gapic_version as package_version
+
+__version__ = package_version.__version__
+
```
Is this generated or written?  If written consider taking it out.  The `gapic_version.__version__` is currently a placeholder at `0.0.0` so there is no   information.

Also this module is a part of the cirq-google package which has its own version at `cirq_google.__version__`.  I don't see a reason to define and maintain a separate version value here.

hoisinberg replied:
It is generated, though I can update that version number if you like. Perhaps to have it reference `cirq_google.__version__`?

pavoljuhas replied:
Please delete it.  It is not there in the `main` so there is no way any code needs it.

pavoljuhas reviewed (changes requested):
If feasible, please deprecate instead of remove the  `whitelisted_users` argument.  Also please see inline suggestions.

hoisinberg commented:
> Is the code in this PR entirely generated or are there some manual changes too?
> 
> If there are some manual changes, the best is to split it to 2 commits, the first with the generated output and a second one with manual changes. This would make it easier to review just the manually-written part as it is not the best use of time to check a generated code.
> 
> Also as a general rule, when you update your PRs (especially after requesting a review) please just add new commits instead of amending and force-pushing. It will make it easier to see changes w/r to the last round of review. Also to sync with main, please just do a merge commit instead of a rebase-on-main.
> 
> The Cirq PRs are squashed to a single commit when merged to the main, but it is still useful to have a more detailed history in PR commits. (These are BTW available from the repo at `refs/pull/7500/head`)

It is nearly all auto-generated. The only manual changes I added were to address CI breakages:

- The linter generated 1400 errors, ~1100 of which were auto-fixed by the linter itself. The remaining 300 or so were all "line too long" complaints which I corrected myself.
- Renaming "whitelisted_users" to "allowlisted_users" throughout the library. This is because the corresponding field on the `QuantumReservation` proto was also renamed from "whitelisted_users" to "allowlisted_users" so attempts to set the now non-existent proto field was causing errors. The internal proto change has been live since 2023 and hasn't caused any breakages for our users, I believe the renaming is safe.
- Updating the failing notebook tests.

pavoljuhas reviewed (approved):
I have added deprecation support for the old `whitelisted_users` arguments in the manually written classes, which should be less disruptive for possible users (09ff1d5246bb39734d86688aad8304d2295735f4).  

LGTM on my part.  I am holding on with merge in case you have some comments or suggestions.
